internal/cmd/keys.go

```diff
@@ -148,10 +148,19 @@ func newKeysRemoveCmd(cli *CLI) *cobra.Command {
 				return err
 			}
 
+			config, err := currentHostConfig()
+			if err != nil {
+				return err
+			}
+
 			logging.Debugf("call server: list access keys named %q", args[0])
 			keys, err := client.ListAccessKeys(api.ListAccessKeysRequest{Name: args[0]})
 			if err != nil {
-				return handleListKeysMissingPrivilege(err)
+				var err2 error
+				keys, err2 = client.ListAccessKeys(api.ListAccessKeysRequest{UserID: config.UserID, Name: args[0]})
+				if err2 != nil {
+					return handleListKeysMissingPrivilege(err)
+				}
 			}
```

Comment:
Is it always correct to assume that the error is related to authorization based on identity?

I think an admin could be removing a key for a different user, and get an error for some other reason.

I wonder if we could improve our API to remove the need to list the keys first. We have a unique identifier for the key, it should be possible to delete it based on that identifier.